### ISSUE TITLE
Inconsistent conditional result types (with more than one zone)

### ISSUE THREAD

**Issue by Roxyrob** [terraform-aws-modules/terraform-aws-route53#73]
## Description

I get same issue as #39 and #47 but for `zone`. It works with only one zone defined, adding a second zone in a simple configuration

Returns:

<pre>
│ Error: Inconsistent conditional result types
│
│   on ..../modules/zones/main.tf line 2, in resource "aws_route53_zone" "this":
│    2:   for_each = var.create ? var.zones : <b>tomap({})</b>
│     ├────────────────
│     │ var.zones is object with 2 attributes
│
│ The true result value has the wrong type: element types must all match for
│ conversion to map.
</pre>

## Versions

- Module version: 2.8.1

- Terraform version: 1.2.2

- Provider version(s): registry.terraform.io/hashicorp/aws v4.12.0

## Reproduction Code

```
module "route53_hz_root" {
  source  = "terraform-aws-modules/route53/aws//modules/zones"
  version = "2.8.1"

  zones =  {
    "domain1.net" = {
      comment = "Production Base hosted zone"
      tags    = {
        "Environment" = "prod"
      }
    }
    "domain2.net" = {
      comment = "Production Base hosted zone"
      tags    = {
        "Environment" = "prod"
      }
    }
  }

  tags = {
    Terraform = "true"
  }
}

```

step1: apply this config with one zone:

```
module "route53_hz_root" {
  source  = "terraform-aws-modules/route53/aws//modules/zones"
  version = "2.8.1"

  zones =  {
    "domain1.net" = {
      comment = "Production Base hosted zone"
      tags    = {
        "Environment" = "prod"
      }
    }
  }

  tags = {
    Terraform = "true"
  }
}
```
step2: Then change as above (Reproduction Code) and apply.

## Additional context

Probably also for `zones` will work the same solution as `records`: changing **tomap({})** to **{}** will solve the issue.

Note: wrapping `var.zone` inside tomap({...}) allow to create hosted zones but this is not a clean solution. Not sure if tied to this issue but every change in any of the zone destroy and recreate every subzones based on root zones also if zone id remains the same.

**Comment by antonbabenko** [comment]
This issue has been resolved in version 2.8.2 :tada:

**Comment by antonbabenko** [comment]
I could not reproduce this issue with the provided steps but nevertheless I have fixed `for_each` in #74 . Please try the updated version - v2.8.2.

**Comment by Roxyrob** [comment]
Hi @antonbabenko,
this worked for me. Thak you !
